client/stripe-utils/utils.js

```diff
@@ -1,7 +1,8 @@
+/* global wc_stripe_upe_params, wc */
+
 /**
  * External dependencies
  */
-import { getSetting } from '@woocommerce/settings';
```

Comment:
I removed that `import` because internally it relies on the global `wc` variable, which is not defined when using the shortcode checkout (and now we are using the same `stripe-utils` files for both shortcode and blocks):

<img width="488" alt="Screen Shot 2021-08-08 at 23 07 44" src="https://user-images.githubusercontent.com/407542/128653999-d7816105-6663-48ca-bba0-ec0cc45f23e6.png">

Is the `wc` global supposed to be defined for the shortcode checkout? maybe I'm missing a js enqueue?

Reply:
> Is the `wc` global supposed to be defined for the shortcode checkout? maybe I'm missing a js enqueue? 

I thought it would be defined there as well, but you're right; it's not present. You're not missing a js enqueue 🙂 

Since that's the case, can we instead create a utility function that's a wrapper around us accessing the `wc` global? That way we can contain the global accessors to one file.

Something along the lines of:

```js
/* global wc_stripe_upe_params, wc */

export const getSettings = () => {
	return 'undefined' !== typeof wc_stripe_upe_params
		? wc_stripe_upe_params
		: wc.wcSettings.getSetting( 'stripe_data', null );
}
```

Reply:
Also, we should probably use optional chaining to access the `wc` global:

```js
wc?.wcSettings?.getSetting( 'stripe_data', null );
```

Reply:
Added chaining to `wc` and `wcSettings()`

Reply:
>Added chaining to `wc` and `wcSettings()`

🎉 

Can we add a wrapper function for retrieving the settings to limit the scope of where we access global values? Maybe this could be part of `getStripeServerData()`?